node/resolvers/profile/index.ts

```diff
@@ -53,6 +83,18 @@ export const mutations = {
     { email, fields, isNewsletterOptIn }: SubscribeNewsletterArgs,
     context: Context
   ) => {
+    const { storeUserAuthToken } = context.vtex
+
+    if (!storeUserAuthToken) {
+      throw new AuthenticationError('Unauthorized')
+    }
+
```

Comment:
Shouldn't be this inside `checkUserAuthorization`? I think it makes more sense since the function would become more flexible and would contain every step needed for completing the task, making the code cleaner.

Reply:
Nice catch